sycl/test/lit.cfg.py

```diff
@@ -200,13 +203,24 @@ def getDeviceCount(device_type):
     lit_config.warning("Accelerator device not found")
 
 if xocc != "off":
+    # xrt doesn't deal well with multiple executable using it concurrently.
+    # so we wrap every use of XRT inside an file lock
+    # we also wrap invocation of executable in an setsid to prevent
+    # a single program faillure from ending all the tests.
```

Comment:
```suggestion
    # XRT doesn't deal well with multiple executable using it concurrently.
    # So we wrap every use of XRT inside an file lock.
    # We also wrap invocation of executable in an setsid to prevent
    # a single program faillure from ending all the tests.
```
Thanks for the very helpful comments!
@stsoe: is this a bug in our SYCL runtime about not using correctly XRT (because for example we should specify some exclusive device use) or XRT does not allow this use case?

Reply:
@keryell XRT is multi-process safe, maybe I am not understand the comment correctly?

Reply:
Thanks @stsoe.
So if XRT is multi-process-safe, I guess we should be able to run all the unit tests on our U200 in parallel without any failure since I assume there would be some kind of serialization around the access of the U200...
Curious. Either a bug in the SYCL runtime or in XRT. @Ralender can you dive a little bit in what happens here?

Reply:
XRT works closely like described here: https://xilinx.github.io/XRT/master/html/multiprocess.html.
I am not sure what you mean by serialization?  Maybe you will be surprised to know:

A process P2 will fail to download a new xclbin if some other process has the current xclbin locked.  The process P2 will fail, it will not be serialized by XRT.  

A process P1 will fail to acquire a lock on a CU after successful xclbin download, if in the mean time another process P2 has successfully downloaded a new xclbin.   P1 will fail, it will not wait for P2 to finish.

Of course there is serialization when all processes agree on the current xclbin and all processes agree to share at least some of the compute units such that no one gets left behind with nothing to execute on.

Reply:
@stsoe Thanks! This is super helpful! That makes sense to have such an execution model on FPGA.
Obvious in our case all the unit tests runs different kernels, so the solution of @Ralender makes sense but the comment in the code above has to be updated by describing a little bit the XRT execution model here.
We probably need to think about extending the SYCL runtime with this knowledge at some point.
And obviously read again the XRT documentation from time to time to be aware of the new features... :-)

Reply:
in https://xilinx.github.io/XRT/master/html/multiprocess.html
it is said "Emulation flow does not have support for multi-process yet."
and tests have to support emulation. 
it is very possible that even if XRT supports multi-process in hw mode.
XRT will report error when there is conflict on a resource and SYCL but there is no code to handle that error which causes test failure.